24.10版本下编译passwall出现libc::getrandom引用错误

### Describe the bug

涉及组件为shadowsocks-rust
此为编译报错：
error[E0425]: cannot find value `getrandom` in crate `libc`
  --> /home/yanmo/CodeSpace/Repositories/OpenwrtLede/immortalwrt/dl/cargo/registry/src/index.crates.io-1949cf8c6b5b557f/getrandom-0.3.2/src/backends/linux_android_with_fallback.rs:34:39
   |
34 |         let fptr: GetRandomFn = libc::getrandom;
   |                                       ^^^^^^^^^ not found in `libc`

For more information about this error, try `rustc --explain E0425`.
error: could not compile `getrandom` (lib) due to 1 previous error

### ImmortalWrt version

b91395a023b6fc9d31d4f538a71fb2f97083e1c7

### ImmortalWrt release

24.10

### ImmortalWrt target/subtarget

ramips/mt7621

### Device

XiaoMi CR6609

### Image kind

Self-built image

### Steps to reproduce

拉取24.10版本, 在make menuconfig中添加luci-passwall组件，后执行make -j10编译即可复现.

### Actual behaviour

_No response_

### Expected behaviour

_No response_

### Additional info

我会尝试自身修复此BUG问题，但没法保证时期。

### Diffconfig

```text

```

### Terms

- [x] I am reporting an issue for ImmortalWrt, not an unsupported fork.

--- a/net/shadowsocks-rust/Makefile
+++ b/net/shadowsocks-rust/Makefile
@@ -6,12 +6,12 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=shadowsocks-rust
-PKG_VERSION:=1.23.0
+PKG_VERSION:=1.23.1
 PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=https://codeload.github.com/shadowsocks/shadowsocks-rust/tar.gz/v$(PKG_VERSION)?
-PKG_HASH:=13307594159cfe23f91e69c8b08ba7a41a17a2f36a4bfb3821476026f7518cac
+PKG_HASH:=0af223694de7261d4cf00f49e34f6dfe9ea5385bfe52e64f96cb0d11f3b947b0
 
 PKG_MAINTAINER:=Tianling Shen <cnsztl@immortalwrt.org>
 PKG_LICENSE:=MIT

